File: completed/1399.swift
class Solution {
    func countLargestGroup(_ n: Int) -> Int {
        var groupSizes: [Int: Int] = [:]
        for i in 1 ... n {
            var sum = 0
            var num = i
            while num > 0 {
                sum += num % 10
                num /= 10
            }
            groupSizes[sum, default: 0] += 1
        }

        var maxSize = groupSizes.values.reduce(0, { x, y in
            max(x, y)
        })
        var count = 0
        return groupSizes.values.reduce(0, { x, y in
            x + (y == maxSize ? 1 : 0)
        })
    }
}

File: completed/1399.cpp
class Solution {
private:
    int digitSum(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

public:
    int countLargestGroup(int n) {
        int largest = 0;
        int count = 0;
        unordered_map<int, int> counts;
        for (int i = 1; i <= n; i++) {
            int sum = digitSum(i);
            counts[sum]++;
            if (counts[sum] > largest) {
                largest = counts[sum];
                count = 1;
            } else if (counts[sum] == largest) {
                count++;
            }
        }
        return count;
    }
};
